code-challenges.test.js

```diff
@@ -16,52 +16,105 @@
 
 // --------------------1) Create a function that takes in an array, removes the first item from the array and shuffles the remaining content.
 
-// a) Create a test with an expect statement using the variable provided.
 
-var colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
-
-
-
-
-// b) Create the function that makes the test pass.
-
-
-
-
-
-// --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
-
-// a) Create a test with expect statements for each of the variables provided.
-
-var nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
-var nums2 = [109, 5, 9, -59, 8, 24]
-// Expected output: [-59, 109]
-
-
-
-
-
-// b) Create the function that makes the test pass.
-
-
-
-
-
-// --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
-
-// a) Create a test with an expect statement using the variables provided.
-
-var testArray1 = [3, 7, 10, 5, 4, 3, 3]
-var testArray2 = [7, 8, 2, 3, 1, 5, 4]
-// Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
-
-
-
-
-
-// b) Create the function that makes the test pass.
+// iterate through the array backwards 
+//create a new, empty array to store all the words that are shuffled
+
+
+// shuffle the words
+// use a conditional to check if the array is shuffled
+// return the new originally blank array
+// call my function
+
+
+let arry1 = ["purple", "blue", "green", "yellow", "pink"]
+let arrayShuffle = function(arr) {
+   for (let i = arr.length -1; i> 0; i--){
+   let newPos = Math.floor(Math.random() * (i + 1));//
+       temp = arr[i]; // declare a variable. temp is equal to arr[i].
+       arr[i] = arr[newPos]; // place the new pos 
+       arr[newPos] = temp; // place what he have in temp, swap which is what the loop is doing
+    }
+   return arr;// returning array
+}
+ let newArray = arrayShuffle(arry1);// testing array to see if the array will shuffle
+console.log(newArray)
```

Comment:
This is a good shuffle function! I appreciate the checking aspect that makes sure the shuffle worked. This prompt called for the array to have the first element removed before a shuffle. Just like removing a flash card once I get it right and shuffling the rest of the flash cards. What kind of changes would need to be made to make this happen? Feel free to reply as a comment to this